src/main/java/javaslang/collection/Array.java

```diff
@@ -203,6 +203,42 @@ private Array(Object[] back) {
         return Array.ofAll(Iterator.ofAll(array));
     }
 
+    /**
+     * Returns an Array containing {@code n} values of a given Function {@code f}
+     * over a range of integer values from 0 to {@code n - 1}.
+     *
+     * @param <T> Component type of the Array
+     * @param n The number of elements in the Array
+     * @param f The Function computing element values
+     * @return An Array consisting of elements {@code f(0),f(1), ..., f(n - 1)}
+     * @throws NullPointerException if {@code n} or {@code f} are null
+     */
+    public static <T> Array<T> fill(Integer n, Function<Integer, ? extends T> f) {
```

Comment:
I just realized that Scala has a different name for `fill(int, Function)`. It is `tabulate`:

```scala
scala> List.tabulate(5)(n => n * n)
res0: List[Int] = List(0, 1, 4, 9, 16)
```

We should rename the methods accordingly because we align to Scala syntax.

Note: The name `fill(int, Supplier)` remains.